### ISSUE TITLE
Must support codec use

### ISSUE THREAD

**Issue by wiibaa** [logstash-plugins/logstash-output-file#7]
*From [LOGSTASH-1795](https://logstash.jira.com/browse/LOGSTASH-1795)*

Using following config:
````
output {
  if [type] == 'apache_access' {
    file {
      path           => "/tmp/logstash.out"
      codec          => oldlogstashjson {}
    }
    file {
      path           => "/tmp/logstash.out.json"
      codec          => json {}
    }
    file {
      path           => "/tmp/logstash.out.plain"
      codec          => plain {}
    }
  }
}
````
Expected: each event written in 3 different formats.
Actual result: all written in (new) json format.

**Comment by EdHurtig** [comment]
:+1: I just ran into this today when using the file output.  This doesn't seem like it should be that hard... Will take a look and see

**Comment by purbon** [comment]
Fixed with #9
